tensorboard/plugins/graph/README.md

```diff
@@ -0,0 +1,129 @@
+```
+
+might be represented as a graph with
+
+-   3 nodes: w, sqrt, result
+-   2 edges: [w, sqrt], [sqrt, result]
+
+Frameworks such as TF2 allow users to manually annotate functions with the
+`@tf.function` decorator to indicate to TensorFlow that a graph should be
+created. Other frameworks may require users to explicitly call a utility
+function to write an implicitly created graph.
+
+Notably, the graph itself does not contain variable values (e.g. weights),
+assets, or signatures (e.g. input and output values).
+
+### Creating a graph
+
+For details guides on creating graphs, please see the docs for
+[TensorFlow](https://www.tensorflow.org/guide/intro_to_graphs) or
+[PyTorch](https://pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter.add_graph)
+
+**Saving to an event file**
+
+In TensorFlow 2, wrap a `@tf.function` decorated function with
+`tf.summary.trace_on(graph=true)` and `tf.summary.trace_export()`. This will add
+the graph data to the event files in the log directory.
+
+```python
+@tf.function
+def my_func():
+  return tf.random.uniform((3, 3))
+
+writer = tf.summary.create_file_writer('logs')
+tf.summary.trace_on(graph=True)
+
+my_func()
+
+with writer.as_default():
+  tf.summary.trace_export(name="my_func", step=0)
+```
+
+In TensorFlow 1, the graph of a session is manually added with `add_graph()`.
+
+```python
+writer = tf.summary.FileWriter('logs')
+with tf.Session() as sess:
+  a = tf.placeholder(tf.float32, shape=(1, 2))
+
+with writer.as_default():
+  writer.add_graph(sess.graph)
+```
+
+**Saving to Protobuf text format (.pbtxt)**
+
+When a `tf.Graph` or `tf.compat.v1.GraphDef` object is available, it can be
+written directly to a file using TensorFlow's
+[`tf.io.write_graph()`](https://www.tensorflow.org/api_docs/python/tf/io/write_graph),
+like so:
+
+```python
+tf.io.write_graph(graph_or_graphdef, '/tmp/dir', 'my_graph.pbtxt')
+```
+
+A `@tf.function` function's graph can be accessed via
+`my_func.get_concrete_function().graph`.
+
+## Using the dashboard
+
+Graphs can be loaded into TensorBoard by either:
+
+-   Launching `tensorboard --logdir /dir/containing/written_graphs` at a logdir
+    with event files containing a graph.
+-   Launching TensorBoard, opening the "Graphs" tab, and clicking "Choose file"
+    in the left sidepane to upload a *.pbtxt from the filesystem.
+
+The dashboard offers a variety of features, including:
+
+**Inspection**
+
+-   Viewing details, attributes of a selected node by clicking it.
+-   Searching for a node by name.
+-   A "Trace inputs" mode to highlight all nodes that may have some effect on
+    the selected node.
+-   Node coloring modes: by structure, device, TPU compatibility, etc.
+-   Indicating "size" of output tensors visually with thicker edges.
+
+**Organization**
+
+-   Expand and collapse group nodes.
+-   Organize nodes automatically by name. TensorBoard uses the "/" forward slash
+    in node names to determine node groups.
```

Comment:
I think it's better to align the items as either starting with verb or noun

Reply:
Done.